cli/cmd/initnodes.go

```diff
@@ -398,3 +423,96 @@ func gethInit(ctx context.Context, cfg InitConfig, dir string) error {
 
 	return nil
 }
+
+func restoreSnapshotData(ctx context.Context, homeDir string, clientName string) error {
+	log.Info(ctx, fmt.Sprintf("⌛ Restoring latest chain snapshot for %s...", clientName))
+
+	clientStorageDir, err := getClientStorageDir(clientName)
+	if err != nil {
+		return err
+	}
+
+	absClientStorageDir := filepath.Join(homeDir, clientStorageDir)
+	origAbsClientStorageDir := fmt.Sprintf("%s_orig_%d", absClientStorageDir, time.Now().Unix())
+
+	// Add _orig_<unix_timestamp> suffix to client storage dir to keep it as a backup.
+	log.Info(ctx, fmt.Sprintf("Renaming %s to %s to keep as backup...", absClientStorageDir, origAbsClientStorageDir))
+	if err := os.Rename(absClientStorageDir, origAbsClientStorageDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("❌ Error renaming %s to %s to keep as backup", absClientStorageDir, origAbsClientStorageDir))
+	}
+
+	// Decompress backup snapshot archive file.
+	snapshotArchive := getSnapshotBackupArchive(clientName)
+	absSnapshotOutputDir := filepath.Join(homeDir, clientName)
+	if err := archive.DecompressTarLz4(ctx, snapshotArchive, absSnapshotOutputDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("❌ Error restoring latest %s snapshot", clientName))
+	}
+
+	// Cleanup backup snapshot archive file.
+	if err := os.Remove(snapshotArchive); err != nil {
+		return errors.Wrap(err, "❌ Error removing latest snapshot backup archive: "+snapshotArchive)
+	}
+
+	switch clientName {
+	case gethClientName:
+		// Copy the newly generated jwtsecret file from the renamed geth home path to the actual geth home path with the newly restored snapshot data.
+		jwtSecretSrcFilePath := origAbsClientStorageDir + "/jwtsecret"
+		jwtSecretDestFilePath := absClientStorageDir + "/jwtsecret"
```

Comment:
please create an issue if we are leaking secrets via our snapshots

Reply:
This `jwtsecret` file is not in the snapshots, it is generated on the fly when running `init-nodes`  before we decompress the snapshot archives, so we just copy it over to the new client storage directory. Same with `nodekey`.

Reply:
do snapshot download BEFORE jwtsecret and other stuff, then no need to worry about this

Reply:
Done, eliminated the need to copy around files and persist backup copies of the configs.